openedx_lti_tool_plugin/resource_link_launch/views.py

```diff
@@ -0,0 +1,409 @@
+    ) -> Tuple[HttpResponse, str]:
+        """Get launch response.
+
+        Args:
+            request: HTTPRequest object.
+            edx_user: User instance.
+            course_id: Course ID string.
+            usage_key_string: Usage key string of a unit or component.
+
+        Returns:
+            A tuple containing a HTTP 302 response corresponding to the
+            requested launch resource (course, unit or component) and a context key.
+
+        """
+        # Get course launch response.
+        if not usage_key_string:
+            context_key = course_id
+            response = self.get_course_launch_response(course_id)
+        # Get unit/component launch response.
+        else:
+            context_key = usage_key_string
+            response = self.get_unit_component_launch_response(usage_key_string, course_id)
+        # Set JWT authentication cookies to launch response.
+        return set_logged_in_cookies(request, response, edx_user), context_key
+
+    def get_course_launch_response(self, course_id: str) -> HttpResponseRedirect:
+        """Get course launch response.
+
+        Args:
+            course_id: Course ID string.
+
+        Returns:
+            HttpResponseRedirect to learning MFE course URL.
+
+        Raises:
+            LtiToolLaunchException: If ALLOW_COMPLETE_COURSE_LAUNCH is disabled.
+
+        """
+        if not ALLOW_COMPLETE_COURSE_LAUNCH.is_enabled():
+            raise LtiToolLaunchException(_('Complete course launches are not enabled.'))
+
+        return redirect(
+            f'{configuration_helpers().get_value("LEARNING_MICROFRONTEND_URL", settings.LEARNING_MICROFRONTEND_URL)}'
+            f'/course/{course_id}'
+        )
+
+    def get_unit_component_launch_response(
+        self,
+        usage_key_string: str,
+        course_id: str,
+    ) -> HttpResponseRedirect:
+        """Get unit or component launch response.
+
+        Args:
+            usage_key_string: Usage key string of a unit or component.
+            course_id: Course ID string.
+
+        Returns:
+            HttpResponseRedirect to `render_xblock` path.
+
+        Raises:
+            LtiToolLaunchException: If UsageKey `course_key` attribute is not equal to
+                the `course_id` value or `usage_key_string` has an incorrect value.
+
+        """
+        usage_key = UsageKey.from_string(usage_key_string)
+
+        if str(usage_key.course_key) != course_id:
+            raise LtiToolLaunchException(_('Unit/component does not belong to course.'))
+
+        if usage_key.block_type in ['chapter', 'sequential', 'course']:
+            raise LtiToolLaunchException(_(f'Invalid XBlock type: {usage_key.block_type}'))
+
+        return redirect('render_xblock', usage_key_string)
+
+    def handle_ags(  # pylint: disable=inconsistent-return-statements
+        self,
+        launch_message: DjangoMessageLaunch,
+        launch_data: dict,
+        lti_profile: LtiProfile,
+        context_key: str,
+    ):
```

Comment:
@anfbermudezme I will keep the return type hint the same, the method doesn't return the `LtiGradedResource` instance, and since the method always returns None the type hint is not necessary, however I noticed the return on the docstring was incorrect so I will fix that